x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsPendingDeletionsIntegTests.java

```diff
@@ -0,0 +1,575 @@
+
+    public void testSearchableSnapshotIsDeletedWithOnRepoCleanUp() throws Exception {
+        mountIndexThenExecute((repository, snapshot, index) -> {
+            try {
+                final int garbageFiles = between(1, 10);
+                final int garbageSize = between(1, 50);
+                final PlainActionFuture<Collection<Void>> garbageFuture = PlainActionFuture.newFuture();
+                final GroupedActionListener<Void> garbageGroupedListener = new GroupedActionListener<>(garbageFuture, garbageFiles);
+
+                final BlobStoreRepository blobRepository = getRepositoryOnMaster(repository);
+                for (int i = 0; i < garbageFiles; i++) {
+                    int garbageId = i;
+                    blobRepository.threadPool()
+                        .generic()
+                        .execute(
+                            ActionRunnable.run(
+                                garbageGroupedListener,
+                                () -> blobRepository.blobStore()
+                                    .blobContainer(blobRepository.basePath())
+                                    .writeBlob("snap-" + garbageId + ".dat", new BytesArray(randomByteArrayOfLength(garbageSize)), true)
+                            )
+                        );
+                }
+                garbageFuture.get();
+
+                blockMasterOnWriteIndexFile(repository);
+
+                final ActionFuture<CleanupRepositoryResponse> cleanUpFuture = client().admin()
+                    .cluster()
+                    .prepareCleanupRepository(repository)
+                    .execute();
+
+                final String masterNode = internalCluster().getMasterName();
+                awaitClusterState(
+                    state -> state.custom(RepositoryCleanupInProgress.TYPE, RepositoryCleanupInProgress.EMPTY).hasCleanupInProgress()
+                );
+                waitForBlock(masterNode, repository);
+
+                assertAcked(client().admin().indices().prepareDelete(mountedIndex(index)));
+                awaitSnapshotPendingDeletion(snapshot);
+                assertFalse(cleanUpFuture.isDone());
+
+                unblockNode(repository, masterNode);
+                awaitNoMoreSnapshotsDeletions();
+
+                final CleanupRepositoryResponse cleanUpResponse = cleanUpFuture.get();
+                assertThat(cleanUpResponse.result().blobs(), equalTo((long) garbageFiles));
+                assertThat(cleanUpResponse.result().bytes(), equalTo((long) garbageSize * garbageFiles));
+
+                expectThrows(
+                    SnapshotMissingException.class,
+                    () -> client().admin().cluster().prepareGetSnapshots(repository).setSnapshots(snapshot.getName()).get()
+                );
+            } catch (Exception e) {
+                throw new AssertionError(e);
+            }
+        });
+    }
+
+    public void testSearchableSnapshotIsDeletedAfterExpiration() throws Exception {
```

Comment:
Can we squeeze in "pending deletion" in the name, since it is the pending deletion that is deleted after expiration?

Reply:
Sure, I renamed to `testSnapshotDeletionsPendingIsRemovedFromClusterStateAfterExpiration` in dd1ced9a409ad9490c9e435b674fa3608eebc58a